hipcheck/src/plugin/types.rs

```diff
@@ -275,34 +331,66 @@ impl TryFrom<Query> for PluginQuery {
 	}
 }
 
-#[derive(Debug)]
 pub struct MultiplexedQueryReceiver {
-	rx: Streaming<PluginQuery>,
+	rx: QueryStream,
 	backlog: HashMap<i32, VecDeque<PluginQuery>>,
 }
+
+impl std::fmt::Debug for MultiplexedQueryReceiver {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		f.debug_struct("MultiplexedQueryReceiver")
+			.field("rx", &"<receiver>")
+			.field("backlog", &self.backlog)
+			.finish()
+	}
+}
+
+/// Helper type for a stream of query messages.
+///
+/// Note that the inner item is a `Result` because the inner
+/// `query` field on the message can technically be missing.
+///
+/// This case is handled in the `message` method to flatten
+/// that kind of error so code consuming the stream doesn't
+/// have to worry about it.
+type QueryStream = Box<dyn Stream<Item = StdResult<PluginQuery, Status>> + Send + Unpin + 'static>;
+
 impl MultiplexedQueryReceiver {
-	pub fn new(rx: Streaming<PluginQuery>) -> Self {
+	pub fn new(rx: QueryStream) -> Self {
```

Comment:
Was it changes from `buf lint` that caused the need for this to become `Box<dyn Stream>` or was it a deficiency in my existing code?

Reply:
Good catch! This was necessary for compilation at one point during the refactoring, but I just checked and I was able to remove it without issue in the final version.